utils/numberofstepsforeachm.go

```diff
@@ -0,0 +1,53 @@
+package utils
+
+import "math"
+
+// OptimizedGetNumberOfStepsForEachM : This function optimizes the Collatz Conjecture for a given range of numbers
+//	between startValue and endValue and returns the number of steps to reach 1.
+func OptimizedGetNumberOfStepsForEachM(startValue, endValue int) map[int]int {
+	//	startValue : starting number of the range
+	//	endValue : ending number of the range
+
+	allSteps := make(map[int]int)
+
+	for num := startValue; num <= endValue; num++ {
+		countSteps := 0
+
+		m := num
+		for m != 1 {
+			//	Optimization 1:
```

Comment:
Why so much space b/w `//` and `Optimization 1:`?

```suggestion
			// Optimization 1:
```

Reply:
Also fix it for the rest.